packages/@aws-cdk/cloud-assembly-schema/lib/manifest.ts

```diff
@@ -19,7 +20,8 @@ export class Manifest {
    * @param manifest - manifest.
    */
   public static save(manifest: assembly.AssemblyManifest, filePath: string) {
-    fs.writeFileSync(filePath, JSON.stringify(manifest, undefined, 2));
+    const withVersion = { version: Manifest.version(), ...manifest };
```

Comment:
I get the intent, but this does open a possible bug where for some reason the `version` on the manifest differs from `Manifest.version`. I would at least add a validation here, and eventually we should prevent the ability to declare the manifest version at instantiation time.

Reply:
We can switch this around so that version will always come from `Manifest.version()` and cannot be overwritten by the manifest. Would that work?

Reply:
Sure, that would be ideal.

Reply:
done